src/braingeneers/_compat/typing.py

```diff
@@ -0,0 +1,26 @@
+"""
+Copyright (c) 2023 Braingeneers. All rights reserved.
+
+braingeneers: braingeneerspy
+"""
+
+
+from __future__ import annotations
+
+import sys
+
+if sys.version_info < (3, 10):
+    from typing_extensions import TypeAlias
+else:
+    from typing import TypeAlias
+
+if sys.version_info < (3, 11):
+    from typing_extensions import Self, assert_never
+else:
+    from typing import Self, assert_never
+
+__all__ = ["TypeAlias", "Self", "assert_never"]
```

Comment:
Looks like typing is used for some of the code, and we suggest to update the rest of the code to use this when you can.

This module is for backporting in regards of typing... but since we are using python 3.10 here, we don't need this. @uwcdc please remove the `_compat` module and its associated tests below.

@DailyDreaming for your reference, this is a practice for Scientific Python Backporting: https://learn.scientific-python.org/development/patterns/backports/#placement-in-a-file

Reply:
The `_compat` module was removed and its associated tests removed.